shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -634,125 +639,239 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
+    slices_[view_id]->render_into(overlay_canvas);
+    overlay_canvas->RestoreToCount(restore_count);
+
+    // This flutter view is never the last in a frame, since we always submit the
+    // underlay view last.
+    frame->set_submit_info({.frame_boundary = false, .present_with_transaction = true});
+    layer->did_submit_last_frame = frame->Encode();
+
+    did_encode &= layer->did_submit_last_frame;
+    platform_view_layers[view_id] = LayerData{
+        .rect = overlay->second,   //
+        .view_id = view_id,        //
+        .overlay_id = overlay_id,  //
+        .layer = layer             //
+    };
+    surface_frames.push_back(std::move(frame));
     overlay_id++;
   }
 
-  // If a layer was allocated in the previous frame, but it's not used in the current frame,
-  // then it can be removed from the scene.
-  RemoveUnusedLayers();
-  // Organize the layers by their z indexes.
-  BringLayersIntoView(platform_view_layers);
+  background_frame->set_submit_info({.present_with_transaction = true});
+  background_frame->Encode();
+  surface_frames.push_back(std::move(background_frame));
+
   // Mark all layers as available, so they can be used in the next frame.
+  std::vector<std::shared_ptr<FlutterPlatformViewLayer>> unused_layers =
+      layer_pool_->RemoveUnusedLayers();
   layer_pool_->RecycleLayers();
 
-  did_submit &= frame->Submit();
+  auto task = [&,                                                         //
+               platform_view_layers = std::move(platform_view_layers),    //
+               current_composition_params = current_composition_params_,  //
+               views_to_recomposite = views_to_recomposite_,              //
+               composition_order = composition_order_,                    //
+               unused_layers = std::move(unused_layers),                  //
+               surface_frames = std::move(surface_frames)                 //
+  ]() mutable {
+    PerformSubmit(platform_view_layers,        //
+                  current_composition_params,  //
+                  views_to_recomposite,        //
+                  composition_order,           //
+                  unused_layers,               //
+                  surface_frames               //
+    );
+  };
+
+  fml::TaskRunner::RunNowOrPostTask(platform_task_runner_, fml::MakeCopyable(std::move(task)));
+
+  return did_encode;
+}
+
+void FlutterPlatformViewsController::CreateMissingOverlays(
+    GrDirectContext* gr_context,
+    const std::shared_ptr<IOSContext>& ios_context,
+    size_t required_overlay_layers) {
+  TRACE_EVENT0("flutter", "FlutterPlatformViewsController::CreateMissingLayers");
+
+  auto missing_layer_count = required_overlay_layers - layer_pool_->size();
+
+  // If raster thread is merged because we're in a unit test or running on simulator, then
+  // we don't need to post a task to create an overlay layer.
+  if ([[NSThread currentThread] isMainThread]) {
+    // Create Missing Layers
```

Comment:
I'd be tempted to just delete this comment and the equivalent one in the async version below; with the renamed method that's pretty clear. That or make the m and l lowercase and add a period, but I'd just nuke it.

Maybe but a comment above the latch like:
```cc
// If the raster thread isn't merged, create layers on the platform thread and block until complete.
```

Reply:
Done, and refactored this to conditionally wait on the latch instead of duplicating as much code